File: challenges/02_comment_to_code/comment_to_code.js
// Example 1: Simple function with array input
// Function name: calculateAverage
// Input: numbers (array of numbers)
// Output: number (average of the array)
function calculateAverage(numbers) {
    const sum = numbers.reduce((a, b) => a + b, 0);
    return sum / numbers.length;
}

// Example 2: String manipulation
// Function name: reverseAndCapitalize
// Input: text (string)
// Output: reversed and capitalized string
function reverseAndCapitalize(text) {
    return text.split('').reverse().join('').toUpperCase();
}

// Example 3: Object transformation
// Function name: formatPerson
// Input: person (object with name and age)
// Output: formatted string
function formatPerson(person) {
    return `${person.name} is ${person.age} years old`;
}

File: core_challenges/02_comment_to_code/comment_to_code.py
# Example 1: Simple function with array input
# Function name: calculate_average
# Input: numbers (list of numbers)
# Output: float (average of the list)
def calculate_average(numbers):
    return sum(numbers) / len(numbers)

# Example 2: String manipulation
# Function name: reverse_and_capitalize
# Input: text (string)
# Output: reversed and capitalized string
def reverse_and_capitalize(text):
    return text[::-1].upper()

# Example 3: Dictionary transformation
# Function name: format_person
# Input: person (dict with name and age)
# Output: formatted string
def format_person(person):
    return f"{person['name']} is {person['age']} years old"
